src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -173,51 +173,29 @@ static KernelInfoCollection CreateKernelInfos(CompositeKernel kernel)
 
             var valueOptionResult = cmdLineContext.ParseResult.GetValueForOption(valueOption);
 
-            // FIX: (HandleSetMagicCommand) 
-
-            switch (events.Count)
-            {
-                case 0: break;
-                case 1: break;
-                case 2: break;
-                default: break;
-            }
-
-            if (valueOptionResult is { })
-            {
-                if (valueOptionResult.Kernel is { } sourceKernelName)
-                {
-                    switch (sourceKernelName)
-                    {
-                        case "input": break;
-                        case "password": break;
-                        default: break;
-                    }
-                }
-                else
-                {
-                }
-
-                if (valueOptionResult.Kernel is { } sourceValueName)
-                {
-                }
-                else
-                {
-                }
+            var sourceKernel = Kernel.Root.FindKernelByName(valueOptionResult.Kernel);
+
+            ValueProduced valueProduced;
+            if (
+                sourceKernel?.KernelInfo.IsProxy == false
```

Comment:
I don't believe `KernelInfo` can ever be null.

Reply:
if the kernel is not found. that is the case when using @input:x

Reply:
Ah right.